src/Planners/include/BipedalLocomotion/Planners/QuinticSpline.h

```diff
@@ -0,0 +1,86 @@
+/**
+ * @file QuinticSpline.h
+ * @authors Giulio Romualdi
+ * @copyright 2020 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_PLANNERS_QUINTIC_SPLINE_H
+#define BIPEDAL_LOCOMOTION_PLANNERS_QUINTIC_SPLINE_H
+
+#include <vector>
+#include <memory>
+
+#include <Eigen/Dense>
+
+namespace BipedalLocomotion
+{
+namespace Planners
+{
+/**
+ * Quintic spline implement a 5-th order polynomial spline in \$f\mathbb{R}^n\$f.
+ */
+class QuinticSpline
+{
+    /**
+     * Private implementation of the class
+     */
+    struct Impl;
+    std::unique_ptr<Impl> m_pimpl; /**< Private implementation */
+
+public:
+
+    /**
+     * Constructor.
+     */
+    QuinticSpline();
+
+    /**
+     * Destructor.
+     * @note It is required by the PIMPL idiom.
+     */
+    ~QuinticSpline();
+
+    /**
+     * Set the knots of the spline.
+     * @param position position of the knots in \$f\mathbb{R}^n\$f.
+     * @param time vector containing the time instant of the knots.
+     * @return True in case of success, false otherwise.
+     */
+    bool setKnots(const std::vector<Eigen::VectorXd>& position, const std::vector<double>& time);
+
+    /**
+     * Set the initial condition of the spline
+     * @param velocity initial velocity (i.e. first derivative).
+     * @param acceleration initial acceleration (i.e. second derivative).
+     * @return True in case of success, false otherwise.
+     */
+    bool setInitialConditions(Eigen::Ref<const Eigen::VectorXd> velocity,
+                              Eigen::Ref<const Eigen::VectorXd> acceleration);
```

Comment:
As written in [stackoverflow](https://stackoverflow.com/questions/21132538/correct-usage-of-the-eigenref-class)
> Then, there is not much difference between a `const Ref<const T>` and a `const Ref<const T>&` because it is unlikely that the function is called with an existing `Ref<const T>` object, and so a `Ref<const T>` will have to be created in most cases.

Since I don't call the functions on an existing `Ref` object. I choose:
-  `Eigen::Ref<const Eigen::VectorXd>` for readable only object
-  `Eigen::Ref<Eigen::VectorXd>` for writable object

I like the symmetry :smile: 

@traversaro @S-Dafarra what do you think about this?

Reply:
I read ``Eigen::Ref`` as a fancy substitution for ``&``. In addition, it should be a very lightweight class with no dynamic memory allocation. I prefer passing them by copy, so I agree with your choice.